src/pages/home/ReportScreen.js

```diff
@@ -6,49 +6,102 @@ import styles from '../../styles/styles';
-import ONYXKEYS from '../../ONYXKEYS';
+import {fetchActions} from '../../libs/actions/Report';
 
 const propTypes = {
-    // id of the most recently viewed report
-    currentlyViewedReportID: PropTypes.string,
-};
+    /* Navigation api provided by react navigation */
+    navigation: PropTypes.shape({
+        /* Display the drawer programmatically */
+        openDrawer: PropTypes.func.isRequired,
+    }).isRequired,
 
-const defaultProps = {
-    currentlyViewedReportID: '0',
+    /* Navigation route context info provided by react navigation */
+    route: PropTypes.shape({
+        /* Route specific parameters used on this screen */
+        params: PropTypes.shape({
+            /* The ID of the report this screen should display */
+            reportID: PropTypes.string,
+        }).isRequired,
+    }).isRequired,
 };
 
-const ReportScreen = (props) => {
-    const activeReportID = parseInt(props.currentlyViewedReportID, 10);
-    if (!activeReportID) {
-        return null;
+class ReportScreen extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+            error: null,
+        };
     }
 
-    return (
-        <ScreenWrapper
-            style={[
-                styles.appContent,
-                styles.flex1,
-                styles.flexColumn,
-            ]}
-        >
-            <HeaderView
-                reportID={activeReportID}
-                onNavigationMenuButtonClicked={() => Navigation.navigate(ROUTES.HOME)}
-            />
-            <View style={[styles.dFlex, styles.flex1]}>
-                <ReportView reportID={activeReportID} />
-            </View>
-        </ScreenWrapper>
-    );
-};
+    componentDidMount() {
+        this.fetchReport();
+    }
+
+    componentDidUpdate(prevProps) {
+        // Reports changed, reset and load new data
+        if (this.reportID !== prevProps.route.params.reportID) {
+            this.fetchReport();
+        }
+    }
+
+    // Todo: ask why getters aren't on top?
+    get canRenderMainContent() {
+        return this.reportID && !this.state.isLoading && !this.state.error;
+    }
+
+    get reportID() {
+        const params = this.props.route.params;
+        return params.reportID;
+    }
+
+    fetchReport() {
+        if (!this.reportID) { return; }
+
+        console.debug('[ReportScreen] Fetch started: ');
+        this.setState({
+            isLoading: true,
+            error: null,
+        });
+
+        fetchActions(this.reportID)
+            .catch((error) => {
+                // Todo: what should we do if fetching fails for some reason
+                // Todo: see what we currently do and do the same here
+                this.setState({error});
+                console.error(error);
+            })
+            .finally(() => {
+                console.debug('[ReportScreen] Stop loading: ');
+                this.setState({isLoading: false});
+            });
+    }
```

Comment:
I'm still fine tuning this.

My end goal is:
If there's already available data to be rendered, we don't need to wait fetching. It should still run and then if there's new data it will auto update

Looking at how currently the app works - there would always be data in Onyx due to `fetchAll` 
So the waiting here can be removed altogether and just usa a brief loading animation to cover the FlatList rendering behind

Flat list rendering need further optimizations - that's the bottleneck on mobile native and why chat actions appear chunk by chunk - struggles with rendering (a lot better on prod build but still...). Nothing that can be done in this ticket though